src/SarifCli/Rules/DoNotUseFriendlyNameAsRuleId.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.Cli.Rules
+{
+    public class DoNotUseFriendlyNameAsRuleId : SarifValidationSkimmerBase
+    {
+        public override string FullDescription => RuleResources.SV0001_DoNotUseFriendlyNameAsRuleIdDescription;
+
+        /// <summary>
+        /// SV0001
+        /// </summary>
+        public override string Id => RuleId.DoNotUseFriendlyNameAsRuleId;
+
+        protected override IEnumerable<string> FormatIds
+        {
+            get
+            {
+                return new string[]
+                {
+                    nameof(RuleResources.SV0001_DefaultFormatId)
+                };
+            }
+        }
+
+        protected override void Analyze(Rule rule, string rulePointer)
+        {
+            if (rule.Id != null
+                && rule.Name != null 
+                && rule.Id.Equals(rule.Name, StringComparison.OrdinalIgnoreCase))
+            {
+                LogResult(
```

Comment:
>LogResult [](start = 16, length = 9)

Up to you. But I always add a source comment when firing messages that shows the complete string that's emitted. Very useful for understanding rule function. See binskim for examples of this discipline.